### PR TITLE
Modern CMake and install step added

### PR DESCRIPTION
**Pull request by Twon**
Update the CMake to use the modern targets based approach to CMake.  This provides an interface target which other users can depend upon in their own project's CMake code to get the targets setting (include directory, required version of C++, compiler flags, and header files).

Additionally, this adds an install step which allows a packaged structure of the project to be produced.  This will allows automatic Zip generation of the project via CPack and will be used when we add a Conan package to generate the package internals providing numerous options for how other users can get access to this library.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,3 +16,28 @@ The linear algebra proposal is targeted for C++23.  Current compiler support is
...
+
+# Building Manually Via CMake
+
+The project can be build via CMake as follows:
+```bash
+cd <project root>
+mkdir build
+cd build
+
+cmake -G <generator> <configuration options> ../linear_algrbra/code
```
Comment (BobSteagall) [inline README.md]: spelling of "algebra"

File: README.md [hunk 2]
```diff
@@ -16,3 +16,28 @@ The linear algebra proposal is targeted for C++23.  Current compiler support is
...
+# Installing Via CMake
+
+Installing the project can be run as follows:
+```bash
+cd <project root>
+mkdir build
+cd build
+
+cmake -G <generator> <configuration options> -DCMAKE_INSTALL_PREFIX=<install dir> ../
+cmake --build ../linear_algrbra/code --target install
```
Comment (BobSteagall) [inline README.md]: spelling of algebra

File: README.md (hunk 1 above)
Comment (Twon) [reply README.md]: Good catch, fixed

File: README.md (hunk 2 above)
Comment (Twon) [reply README.md]: Fixed

**Comment by Twon** [comment]
> In general, I'm OK with the proposed changes (modulo spelling typos mentioned elsewhere). One concern, however, is requiring such a modern version of CMake. Version 3.14 has only been around for about 8 months; some users may be unwilling/unable to upgrade.
> 
> That being said, if there are complaints or requests to go back to older versions of CMake, we can handle them at that time.

Yes, this is correct.  Thanks for raising this, I do have a plan for this, however, I should provide some more details on this. The impact of the CMake version will only be limited to people who need to build the test suite.  For all other users of the interface library (the linear algebra headers), they will link against this target that is automatically generated as part of the install target (I'll add comments to the code review with references to provide some background information).  The generated target will not export the CMake version (this is the way it works, it's designed to be version agnostic).

As for users who do wish to build the tests, this will be resolved when I add a Conan package.  Part of the Conan package definition is the build_requires tag which allows tools that are part of the build process are installed into a temp directory.  CMake will be handled in this fashion so that even if the user does not have CMake installed or has an old version the build process will always pass because it will be able to sync a version for the build.  This will also allow the Travis and Appveyor config to be cleaned up to not have to explicitly install the version of CMake it requires.

**Pull request merged**